weave-js/src/components/PagePanelComponents/Home/Browse3/feedback/HumanFeedback/tsHumanFeedback.ts

```diff
@@ -0,0 +1,68 @@
+import {useEffect, useMemo, useState} from 'react';
+
+import {useBaseObjectInstances} from '../../pages/wfReactInterface/baseObjectClassQuery';
+import {
+  TraceObjQueryReq,
+  TraceObjSchema,
+} from '../../pages/wfReactInterface/traceServerClientTypes';
+import {projectIdFromParts} from '../../pages/wfReactInterface/tsDataModelHooks';
+import {objectVersionKeyToRefUri} from '../../pages/wfReactInterface/utilities';
+import {
+  HumanAnnotationPayload,
+  tsHumanAnnotationSpec,
+} from './humanFeedbackTypes';
+
+export const useHumanFeedbackSpecs = (
+  entity: string,
+  project: string
+): tsHumanAnnotationSpec[] => {
+  const req: TraceObjQueryReq = {
+    project_id: projectIdFromParts({entity, project}),
+    filter: {
+      latest_only: true,
+    },
+  };
+  const res = useBaseObjectInstances('AnnotationSpec', req);
+  const [cols, setCols] = useState<TraceObjSchema[] | null>(null);
+
+  useEffect(() => {
+    if (res.loading || res.result == null) {
+      return;
+    }
+    setCols(res.result);
+  }, [res.loading, res.result]);
+
+  return useMemo(() => {
+    if (cols == null) {
+      return [];
+    }
+    return cols.map(col => ({
+      ...col.val,
+      // attach object refs to the columns
+      ref: objectVersionKeyToRefUri({
+        scheme: 'weave',
+        weaveKind: 'object',
+        entity,
+        project,
+        objectId: col.object_id,
+        versionHash: col.digest,
+        path: '',
+      }),
+    }));
+  }, [cols, entity, project]);
+};
+
+export const extractValFromHumanAnnotationPayload = (
+  payload: HumanAnnotationPayload
+) => {
+  if (payload.value == null) {
+    return null;
+  }
+  try {
+    const val = Object.values(Object.values(payload.value)[0])[0];
```

Comment:
this smells... feels like there is unnecessary nesting or something

Reply:
this is to unwrap the feedback payloads: 
```value: {
   [annotationSpecName]: {
      [artifactVersion]: value
   }
} ```

which is the type I thought we decided on?

Reply:
That was one iteration. but they are typed as 
```
class AnnotationPayloadSchema(BaseModel):
    value: Any


class RunnablePayloadSchema(BaseModel):
    output: Any
```

The name in the type and the ref in the column provide this information. So no, value should equal value here.

Reply:
this should make everything a lot easier actually

Reply:
I think this makes filtering on the table much harder

Reply:
but yeah, fck i missed that adding the annotation_ref on the table _replaced_ this payload structure rather than adding to it...

Reply:
damn - we should have synced closer on this. Can you share which part makes this harder in terms of filtering? The use case I am thinking of makes this easier

Reply:
Let me weave it through and test it, maybe its not harder, just different.